portal/src/graphql/adminapi/AuditLogScreen.tsx

```diff
@@ -149,13 +174,36 @@ const AuditLogScreen: React.FC = function AuditLogScreen() {
         .toJSDate()
         .toISOString();
     }
-    return lastUpdatedAt;
+    return lastUpdatedAt.toISOString();
   }, [rangeTo, lastUpdatedAt]);
 
+  const location = useLocation();
+
   const isCustomDateRange = rangeFrom != null || rangeTo != null;
 
   const { renderToString } = useContext(Context);
 
+  useEffect(() => {
```

Comment:
Instead of syncing searchParams and state, can we eliminate state and rely only on searchParams as a single source of truth?